pydantic/_internal/_generate_schema.py

```diff
@@ -941,6 +941,7 @@ def _generate_dc_field_schema(
         return core_schema.dataclass_field(
             name,
             common_field['schema'],
+            init=field_info.init,
```

Comment:
```suggestion
            init=field_info.init or True
```
@dmontagu, should this be changed?

Reply:
I believe that if it's not specified it uses the value `None`, which gets removed, and the default value in the rust side is True. So basically if you don't specify it uses the rust-side default, if you do specify it uses what you specified. So basically I don't think we need the `or True` here. If tests fail obviously that's wrong.

I'm okay with either approach but I think it makes sense to just not add it to the schema if it's not specified (i.e., use the default `None` value).

Reply:
Also we definitely shouldn't use `field_info.init or True` because that will just always be True lol. Maybe what you meant was `True if field_info.init is None else field_info.init` but, as noted above, I think even that is unnecessary.

Reply:
Ah yes, I meant the latter. Agreed though - no change is needed bc of the rust default. Resolving this convo!